docs/addons/addon-migration-guide.mdx

```diff
@@ -17,79 +17,120 @@ Begin by updating your Storybook dependencies. Use the `next` tag for pre-releas
 
-To ensure your addon supports both Babel and SWC, you can build a custom bundling plugin using [Unplugin](https://unplugin.vercel.app/) that will work with both Webpack and Vite builders, giving you complete control to run Babel (or whatever you want) on stories and components as they are loaded.
+1. Implement the alias at compile time in your addon
+2. Update your bundling configuration to ensure correct dependencies are used
 
-As a workaround, update your documentation to tell users to opt-in to Babel support. This should fix your addon in their project at the cost of performance:
+### TypeScript Requirements
 
-{/* prettier-ignore-start */}
+Storybook now requires TypeScript 4.9 or later. Ensure your addon is compatible with this version.
 
-<CodeSnippets path="storybook-addon-compiler-babel-auto-install.md" />
+### Sidebar Component Changes
 
-{/* prettier-ignore-end */}
+1. The 'extra' prop has been removed from the Sidebar's Heading component
+2. Experimental sidebar features have been removed:
+   - `experimental_SIDEBAR_BOTTOM`
+   - `experimental_SIDEBAR_TOP`
 
-## Migration Example
+### Type System Updates
 
-The Addon Kit [repository](https://github.com/storybookjs/addon-kit) has already been updated to support Storybook 8.0, and you can use it as a reference for your migration. You'll see the changes mentioned in this guide, including ESM support via the `type: module` configuration. As an addon maintainer, we encourage you to update your addon to include them. It simplifies the setup and makes it easier for users to use your addon with the latest version of Storybook. If you choose to follow along with the ESM migration, we've prepared an abbreviated list of changes below.
+The following types have been removed:
+- `Addon_SidebarBottomType`
+- `Addon_SidebarTopType`
+- `DeprecatedState`
 
-* [`package.json`](https://github.com/storybookjs/addon-kit/compare/79282986..fa7ee7c#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519) for dependency management, ESM support and updates to the addon's entry points.
-* [`tsup.config.ts`](https://github.com/storybookjs/addon-kit/compare/79282986..fa7ee7c#diff-8fed899bdbc24789a7bb4973574e624ed6207c6ce572338bc3c3e117672b2a2) for bundling changes, factoring in Storybook's globals.
-* [`.storybook/local-preset.js`](https://github.com/storybookjs/addon-kit/compare/79282986..fa7ee7c#diff-390b53ea479b1ceffcbf31944f644ee23aa9f337b75a8a0ffd815bed50d376cb) to support the ESM migration.
+## Migration Example
 
-<Callout variant="info" icon="💡">
-  For a complete overview of the changes applied to the Addon Kit to fully support Storybook 8.0, see the following [diff view](https://github.com/storybookjs/addon-kit/compare/79282986..fa7ee7c).
-</Callout>
+For a complete example of an addon updated to support Storybook 9.0, refer to the [Addon Kit repository](https://github.com/storybookjs/addon-kit). It demonstrates all the necessary changes for modern addon development.
```

Comment:
Is there a specific PR we can reference here?

Reply:
Good point! Will add a reference.